### PR TITLE
fix!: use an external git repository

### LINKED ISSUE #449: Cloud Source Repositories API cannot be enabled in new projects
We received reports about the fact that [Cloud Source Repositories](https://cloud.google.com/source-repositories/docs) APIs cannot be enabled in new projects.

We should fix this.

Thanks @laurentgrangeau for working on this!

/cc @grapesfrog

### PR DESCRIPTION
**Pull request by ferrarimarco**
- Don't enable the Cloud Source Repositories API.
- Don't provision a Cloud Source Repository.
- Get the Config Sync Git repository with an input variable.
- Let users specify the credentials type to authenticate Config Sync with their repositories.

Fix #449

/cc @grapesfrog

### CODE REVIEW

**Comment by ferrarimarco** [comment]
@laurentgrangeau Can you please review this one?

Basic Terraform validation (`terraform init`, `terraform validate`) should pass on this one. I'll do a complete end-to-end testing while working on the follow-up PR to simplify the deployment process.

File: terraform/variables.tf [hunk 1]
```diff
@@ -114,20 +114,30 @@ variable "enable_confidential_nodes" {
 }
 
 variable "acm_version" {
-  description = "Anthos Config Management version"
+  description = "Config Management version"
```
Comment (laurentgrangeau) [inline terraform/variables.tf:117]: Is it Config Sync instead ?

**Review by laurentgrangeau — COMMENTED** [review]
Just a question for you

File: terraform/variables.tf (hunk 1 above)
Comment (ferrarimarco) [reply terraform/variables.tf:117]: The provider still says ACM: https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/gke_hub_feature_membership#version

**Review by laurentgrangeau — CHANGES_REQUESTED** [review]
Please merge main into your PR and it should be good to go

**Review by laurentgrangeau — APPROVED** [review]

**Pull request merged**